core-bundle/src/Image/Studio/FigureBuilder.php

```diff
@@ -310,7 +310,7 @@ public function fromImage(ImageInterface $image): self
     /**
      * Sets the image resource by guessing the identifier type.
      *
-     * @param int|string|FilesModel|ImageInterface|null $identifier Can be a FilesModel, an ImageInterface, a tl_files UUID/ID/path or a file system path
+     * @param int|string|FilesModel|ImageInterface|\Throwable|null $identifier Can be a FilesModel, an ImageInterface, a tl_files UUID/ID/path or a file system path
      */
     public function from($identifier): self
```

Comment:
@m-vo is it OK to allow exceptions here?

IMO this is the cleanest way to support `->buildIfResourceExists()` while still letting `->build()` throw a meaningful exception.

Reply:
Or should we better only allow `InvalidResourceException`s to be passed here?

Reply:
Could we instead create an `@internal` setter method `setLastException(InvalidResourceException $exception): void` on the `FigureBuilder`?

Then `createPreviewFigureBuilder()` could initialize the `FigureBuilder` and set an exception in case of an error.

Reply:
Done in bf04b323ef503498121a430617af432f29962f7a